src/lib/comms/sol-lwm2m-client.c

```diff
@@ -1040,6 +1040,13 @@ handle_read(struct sol_lwm2m_client *client,
     uint16_t i;
     int r;
 
+    if ((obj_ctx->obj->id == SECURITY_OBJECT_ID) &&
```

Comment:
Because `handle_read()` can have other entry points, such as `notification_cb()`, which will not pass through `handle_resource()` in case it's called due to `sol_lwm2m_client_notify()`.